components/net/hsts.rs

```diff
@@ -132,3 +148,7 @@ pub fn secure_url(url: &Url) -> Url {
         url.clone()
     }
 }
+
+fn reg_host<'a>(url: &'a str) -> String {
```

Comment:
I don't think this is needed? Just use `reg_suffix`, and when needed do `to_owned` or similar, but you don't need a `String` all the time.